doc/developers/advanced_installation.rst

```diff
@@ -201,6 +201,88 @@ It is however preferred to use pip.
 On Unix-like systems, you can equivalently type ``make in`` from the top-level
 folder. Have a look at the ``Makefile`` for additional utilities.
 
+.. _building_with_meson:
+
+Building with Meson
+-------------------
+
+Supported for Meson is experimental, in scikit-learn 1.5.0.dev0.
+`Open an issue <https://github.com/scikit-learn/scikit-learn/issues/new>`__ if
+you encounter any problems!
+
+Make sure you have `meson-python` installed:
+
+.. code-block:: bash
+
+    conda install -c conda-forge meson-python -y
+
+.. warning::
+
+    Use conda for this, if you install meson-python with pip, this does not not
+    include ninja, and you need another way to install ninja (e.g. package
+    manager)
+
+Simplest way to build with Meson
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To build scikit-learn, the simplest way is to run:
+
+.. code-block:: bash
+
+    make dev-meson
```

Comment:
so this does an editable install. How does one installs in non-editable mode?

Reply:
I would argue that for now editable command is the way to go. As I mentioned a few times, the idea is to have maintainers include Meson in their day-to-day work and improve/tweak things as we go.

Do you have a specific reason in mind, why you would want to use non-editable installs? One way is to use the edit `pyproject.toml` set `build-backend` and use the `pip` command without `--editable`.

Reply:
The idea is to see how the workflow works for us, that includes:

- building and installing with meson
- building for contributors, a bunch of those having windows machines where they don't have `make`

I'm quite uncomfortable with having a python file calling pip. Also, I would like to separate build from install. We should have the equivalent of `pip install -e . --no-build-isolation` as well as `make in`, which do very different things. The former including the latter as a step. Also, a `make clean` shouldn't uninstall the package.

Exactly because the idea is to check our day-to-day work, I'm not comfortable with the current solution.

Reply:
Maybe I was not explicit enough about this and/or it may have been lost in the many comments above. This is the way I see things unfolding (happy to tweak things according to feed-back):
- until we decide otherwise, setuptools remain the main way to build scikit-learn. This is the case even if this PR is merged.
- if this PR is merged, maintainers would be more than welcome to try out Meson through `make dev-meson` and report issues so that we can improve the situation if needed. During this transition period, non-maintainers are not expected to try out Meson. Maintainers that want to keep using setuptools are completely free to do so (i.e. use `make in`, `pip install -e . --no-use-pep517 --no-build-isolation` or `python setup.py develop`)
- hopefully after some time (maybe 1 or 2 months?), most maintainers are convinced that Meson works well and we have fixed teething issues if any arises. We can then switch to Meson as the main build backend, i.e. set `build-backend = "mesonpy"` in `pyproject.toml`. This removes the need from the `build_tools/build-meson-editable-install.py` (I guess this is what you mean by by "a python file calling pip")
- once meson is our main building tool, the doc is updated and all contributors use Meson 

Hopefully, this alleviates some of your concerns :crossed_fingers:!